iroh/src/commands/start.rs

```diff
@@ -124,6 +124,49 @@ where
     Ok(())
 }
 
+/// Migrate the flat store from v0 to v1. This can not be done in the store itself, since the
+/// constructor of the store now only takes a single directory.
+fn migrate_flat_store_v0_v1() -> anyhow::Result<()> {
```

Comment:
can you add a test for this, I know it is annoying but we had issues with migrations a couple of times in the wild now
the main cases to cover are (1) doesn't fail/works on a fresh new setup (2) works with an old version (3) doesn't do anything on a new version

Reply:
I added 2. I think every other cli test test 1. Not sure what to do about 3. I guess I could look at the log and make sure the lines indicating that a migration is happening are not appearing. But that would be super fragile. So short of adding some kind of event for this I would not know how to test this.